odoo/api.py

```diff
@@ -387,6 +387,7 @@ def method(self, args):
 
 def _model_create_single(create, self, arg):
     # 'create' expects a dict and returns a record
+    _create_logger.warning("Rewrite the create method to work in batch for model %s", self._name)
```

Comment:
You should move that warning in the decorator `model_create_single`, so that the warning is only emitted when the method is defined.